### DISCUSSION TITLE
PHOTOPRISM_AUTH_MODE: "public", app keeps asking for a password

### DISCUSSION THREAD

**Discussion by ghost** [photoprism/photoprism#2559] [Q&A] [answered]
Greetings.

I've just pulled the latest version of PhotoPrism, but it isn't public anymore. It keeps asking for credentials.

I've changed the Docker compose file, I've removed  PHOTOPRISM_PUBLIC: "true" and added PHOTOPRISM_AUTH_MODE: "public"

Does somebody knows a way to make it public again?
I'm sharing the photos with the family including IP restriction, so an password isn't needed.

**Comment by lastzero** [comment]
Seems strange as each alone should be enough. Did you restart properly? Our demo is running in public mode, so it should work in general.

**Reply by ghost** [reply]
I've restarted the app indeed.

docker-compose stop
docker-compose up -d

**Reply by lastzero** [reply]
Can you try only using one variable?

**Reply by graciousgrey** [reply]
Just tested this, for me PHOTOPRISM_PUBLIC: "true" does work while PHOTOPRISM_AUTH_MODE: "public" does not work.

**Comment by wechsler42** [comment]
Hi there,
I  second the observations  @PanaromicTiger made.

I read in the release notes:
Breaking Changes
- Config: [PHOTOPRISM_AUTH has been removed in favor of PHOTOPRISM_AUTH_MODE](https://github.com/photoprism/photoprism/commit/591a6562707457045f504defba69e693afccba65)
- Config: [PHOTOPRISM_PUBLIC has been deprecated in favor of PHOTOPRISM_AUTH_MODE](https://github.com/photoprism/photoprism/commit/591a6562707457045f504defba69e693afccba65)

So I deleted PHOTOPRISM_PUBLIC: "true" and added PHOTOPRISM_AUTH_MODE: "public" in my docker-compose file and restarted the app. After that I was asked for credentials although login without authentication was expected. 
In  a next step, I added again  PHOTOPRISM_PUBLIC: "true"  to my docker-compose file leaving PHOTOPRISM_AUTH_MODE: "public"  in place. After restarting the app login without credentials worked as expected.
In a next step I removed PHOTOPRISM_AUTH_MODE: "public"  from docker-compose file just leaving PHOTOPRISM_PUBLIC: "true"  in place. After restarting the app login without credentials worked as expected.

Hope this helps to track down this issue.

**Reply by lastzero** [reply]
Keep using the old variable then. It's just deprecated, not removed.

**Comment by lastzero** [comment]
I see the issue, might be able to fix it tomorrow.

**Reply by ghost** [reply]
I'm happy to read this. When fixed, can I get the fix with docker-compose pull?

**Reply by lastzero** [reply]
Started a new preview build build for testing, available shortly. If it works, we can release it today!

**Reply by lastzero** [reply]
See:
- #2565

**Comment by wechsler42** [comment, marked as answer]
Thanks for the quick fix, with the Build [220730-0e1222c8] of today it works now as intended :-)
So I deleted PHOTOPRISM_PUBLIC: "true" and added PHOTOPRISM_AUTH_MODE: "public" in my docker-compose file and restarted the app. After that login without credentials worked as expected..

**Reply by lastzero** [reply]
Thanks for testing! Glad it's working now 🤗

**Reply by ghost** [reply]
Sorry for the late reply. It is working now! Thanx for the support :)

**Comment by uem56** [comment]


**Reply by ghost** [reply]
I've checked this. PHOTOPRISM_ADMIN_PASSWORD is on the first line of environment:
PHOTOPRISM_AUTH_MODE: "public" is on the third line.

Photoprism is currently public.

**Reply by lastzero** [reply]
- the deprecated `PHOTOPRISM_PUBLIC: "true"` option automatically sets `PHOTOPRISM_AUTH_MODE: "public"` until we remove it in a future release
- `PHOTOPRISM_ADMIN_PASSWORD` should not change the values of `PHOTOPRISM_PUBLIC` or `PHOTOPRISM_AUTH_MODE` in any way

If there's an edge case we haven't considered yet, please let us know! Note, however, that we are currently on vacation, so don't expect a new release this month.
